doc/tb3_tutorial/README.md

```diff
@@ -0,0 +1,573 @@
+
+    source ~/ros2_all_ws/ros2_ws/install/local_setup.bash
+    ros2 run demo_nodes_cpp talker
+
+Open another terminal and run (this node is written in Python)
+
+    source ~/ros2_all_ws/ros2_ws/install/local_setup.bash
+    ros2 run demo_nodes_py listener
+   
+After running both nodes, you should see results similar to this.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/ros2_test.PNG?raw=true)
+
+#  ROS2-Gazebo Installation
+
+We will be using Gazebo with ROS 2, so we need to install the Gazebo packages for ROS2. We will test ROS2-Gazebo after installing Gazebo 9. The fallowing command should install all the necessary ros-gazebo packages for ROS Dashing distribution.
+
+    sudo apt-get install ros-dashing-gazebo*
+
+# Gazebo Installation
+
+We will use Gazebo as our simulator, so let's install Gazebo 9. You can find more information about Gazebo [**here**](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwj_wKXW-vniAhXgJTQIHQmGA2EQFjAAegQIABAB&url=http://gazebosim.org/&usg=AOvVaw3bYv0Z_AA1EEyIXg8sGUuF). 
+
+    curl -sSL http://get.gazebosim.org | sh
+
+If for some reason this doesn't work, you can also use `sudo apt-get install gazebo9` to install Gazebo.
+
+If you see a message saying Gazebo is already installed you can skip this step. That means Gazebo is already installed.
+
+## Test Gazebo
+
+Let's test ROS2-Gazebo and Gazebo together.
+
+    sudo apt install ros-dashing-ros-core ros-dashing-geometry2
+    gazebo --verbose /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_diff_drive_demo.world
+
+If Gazebo prints "plugin can't be found" error message, try running it again
+
+    soruce /opt/ros/dashing/setup.bash
+
+You should see a similar result. Now let's publish to a topic to move our robot.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Screenshot%20from%202019-06-27%2010-08-56.png?raw=true)
+   
+     ros2 topic pub /demo/cmd_demo geometry_msgs/Twist '{linear: {x: 1.0}}' -1
+
+You should see our car moving forward. That means everything is working as expected.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_test.gif?raw=true)
+
+# Navigation 2 Installation Steps
+
+Now, it is time to download and build Navigation2 stack. Stack means that Navigation2 consists of many other packages. By building the stack we will build all this necessary packages for Navigation 2 at once. Please note that we are going to test Navigation2 after building the Turtlebot3 stack.
+
+**Install the dependencies for this repository**
+
+    sudo apt-get install -y \
+        libsdl-image1.2 \
+        libsdl-image1.2-dev \
+        libsdl1.2debian \
+        libsdl1.2-dev
+
+**Clone the Navigation2 Stack in our main workspace.**
+
+    cd ~/ros2_all_ws
+    wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/initial_ros_setup.sh
+    chmod a+x initial_ros_setup.sh
+    ./initial_ros_setup.sh
```

Comment:
This is going to download and build all of ROS 2 all over again. We only need to build it once.

Reply:
You are right. I didn't check what actually happens in the scripts  (initial_ros_setup, build_all). I entirely changed this step. I don't use the scripts anymore. I used the method that you suggested before.

- clone repos
- rosdep
- colcon build

PS: I haven't tested this step yet.

Reply:
I have tested it. The instructions are correct.